Tools/dea_tools/temporal.py

```diff
@@ -434,8 +433,13 @@ def temporal_statistics(da, stats):
         temporal statistics
 
     """
+    # Attempt to import hdstats and raise an error if not available
+    try:
+        import hdstats
+    except ImportError:
+        raise ImportError("`hdstats` is required for `temporal_statistics`. Please install the `[hdstats]` dependency.")
```

Comment:
Its unfortunate that we need `hdstats` in this module when the (more commonly used) phenology and linear regression functions don't require it.

Reply:
Its only the [temporal_statistics](https://github.com/GeoscienceAustralia/dea-notebooks/blob/34d91d531909fb29800d25d0ecfbc29edb0b6c8d/Tools/dea_tools/temporal.py#L399) function that uses hdstats.  Conceptually it makes sense to be in the `temporal.py` file, but if it blocks usage of other, more widely used functions then I don't have a problem with splitting it out. Maybe this function should instead live within `odc-ago` where `hdstats` is already a dependency for the geomedian code-base?

Reply:
Because the module import is now inside the `temporal_statistics` function, I don't think it should block importing other functions from the module any longer - the error will only be returned if `temporal_statistics` is imported without `hdstats` installed.

This is an approach I want to do for `datacube` too - move any of those optional imports to the function level, vs. module level.